src/protagonist/Engine.Tests/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClientTests.cs

```diff
@@ -299,6 +311,106 @@ public async Task GenerateThumbnails_ReturnsThumbForSuccessfulResponse_AfterFirs
             "Landscape images - invalid sizes altered",
         },
     };
+    
+    [Fact]
+    public async Task GenerateThumbnails_UpdatesHandlerWithCookies()
+    {
+        // Arrange
+        var assetId = new AssetId(2, 1, nameof(GenerateThumbnails_ReturnsThumbForSuccessfulResponse));
+        var context = IngestionContextFactory.GetIngestionContext(assetId: assetId.ToString());
+
+        var response = new HttpResponseMessage(HttpStatusCode.OK);
+        response.Headers.Add("Set-Cookie", new List<string?>()
+        {
+            "AWSALB=_remove_; Path=/",
+            "AWSALBCORS=_remove_; Path=/"
+        });
+        httpHandler.SetResponse(response);
+        List<CookieHeaderValue> cookieHeaders = new();
+
+        context.Asset.Width = 2000;
+        context.Asset.Height = 2000;
+
+        context.WithLocation(new ImageLocation
+        {
+            S3 = "//some/location/with/s3"
+        });
+        
+        await sut.GenerateThumbnails(context, defaultThumbs, ThumbsRoot);
+        
+        httpHandler.RegisterCallback(message => cookieHeaders = message.Headers.GetCookies().ToList());
+        httpHandler.GetResponseMessage("{ \"engine\": \"hello\" }", HttpStatusCode.OK);
+    
+        // Act
+        await sut.GenerateThumbnails(context, defaultThumbs, ThumbsRoot);
+
+        // Assert
+        cookieHeaders.Count.Should().Be(2);
+        cookieHeaders[0].Cookies[0].Name.Should().Be("AWSALB");
+        cookieHeaders[0].Cookies[0].Value.Should().Be("_remove_");
+        cookieHeaders[1].Cookies[0].Name.Should().Be("AWSALBCORS");
+    }
+    
+    [Fact]
+    public async Task GenerateThumbnails_DoesNotUpdateHandlerWithCookiesWhenUnrecognised()
+    {
+        // Arrange
+        var assetId = new AssetId(2, 1, nameof(GenerateThumbnails_ReturnsThumbForSuccessfulResponse));
```

Comment:
prefer `AssetIdGenerator`, this is using `nameof()` for an alternative method, which `AssetIdGenerator` will help avoid (same with other methods)